scripts/nn/layers/tanh.dml

```diff
@@ -29,21 +29,13 @@ forward = function(matrix[double] X)
   /*
    * Computes the forward pass for a tanh nonlinearity layer.
    *
-   *   ```
-   *   tanh(x) = (e^x - e^-x) / (e^x + e^-x)
-   *           = 2 * sigmoid(2x) - 1
-   *   ```
-   *
    * Inputs:
    *  - X: Inputs, of shape (any, any).
    *
    * Outputs:
    *  - out: Outputs, of same shape as `X`.
    */
-  # out = (exp(X) - exp(-X)) / (exp(X) + exp(-X))
-  # Simplification of the above formulation to use the sigmoid function:
-  sigma2X = sigmoid::forward(2*X)
-  out = 2*sigma2X - 1
+  out = tanh(X)
```

Comment:
if we're going to swap this out, can you first make a copy of this tanh layer into the `nn/test` folder, and add a unit test comparing the old and new versions in `nn/test/test.dml`?

Reply:
Done.